src/sentry/testutils/silo.py

```diff
@@ -254,33 +269,40 @@ def validate_protected_queries(queries: Iterable[Dict[str, str]]) -> None:
             operation = match.group("operation")
             if operation == "start":
                 fence_depth += 1
+                start_fence_index = index
             elif operation == "end":
                 fence_depth = max(fence_depth - 1, 0)
             else:
                 raise AssertionError("Invalid fencing operation encounted")
 
         for protected in get_protected_operations():
-            if protected.match(sql):
-                if fence_depth == 0:
-                    msg = [
-                        "Found protected operation without explicit outbox escape!",
-                        "",
-                        sql,
-                        "",
-                        "Was not surrounded by role elevation queries, and could corrupt data if outboxes are not generated.",
-                        "If you are confident that outboxes are being generated, wrap the "
-                        "operation that generates this query with the `unguarded_write()` ",
-                        "context manager to resolve this failure. For example:",
-                        "",
-                        "with unguarded_write():",
-                        "    record.delete()",
-                        "",
-                        "Full query log:",
-                        "",
-                    ]
-                    msg.extend([q["sql"] for q in queries if q["sql"]])
-
-                    raise AssertionError("\n".join(msg))
+            if protected.match(sql) and fence_depth == 0:
+                start = max(0, start_fence_index - context_queries)
+                end = min(index + context_queries, len(queries))
+
+                context_queries = queries[start:end]
```

Comment:
I'm assuming with the new fence_depth condition this is only running once, but just double checking that we're okay overwriting the `context_queries` variable from an `int` to a `List` here.

Reply:
Yeah this only runs once but shadowing the variable is a mistake. Thanks for catching that.